frontend/src/features/mapSideBar/hooks/useSideBarBuildingLoader.tsx

```diff
@@ -44,30 +51,36 @@ const useSideBarBuildingLoader = ({
         disabled ? SidebarContextType.VIEW_BUILDING : SidebarContextType.UPDATE_BUILDING,
       );
     };
-    if (showSideBar && buildingId && buildingId !== cachedBuildingDetail?.id) {
+    if (showSideBar && !!buildingId && buildingId !== cachedBuildingDetail?.id) {
       loadBuilding();
-    } else if (!buildingId) {
-      if (!!cachedBuildingDetail) {
-        setCachedBuildingDetail(null);
-      }
+    } else if (!buildingId && buildingId !== 0 && !!cachedBuildingDetail) {
+      setCachedBuildingDetail(null);
+    } else if (!buildingId && associatedBuildingDetail) {
+      setCachedBuildingDetail({
+        ...associatedBuildingDetail.parcelDetail,
+        id: 0,
+        financials: [],
+      } as any);
     }
   }, [dispatch, buildingId, disabled, showSideBar]);
 
   /**
    * Handle the user swapping between view and update of the same property.
    */
   useDeepCompareEffect(() => {
-    if (
-      !!buildingId &&
-      cachedBuildingDetail?.id === buildingId &&
-      [SidebarContextType.VIEW_BUILDING, SidebarContextType.UPDATE_BUILDING].includes(
-        sideBarContext,
-      )
-    ) {
-      const newContext = disabled
-        ? SidebarContextType.VIEW_BUILDING
-        : SidebarContextType.UPDATE_BUILDING;
-      newContext !== sideBarContext && setSideBarContext(newContext);
+    if (!!buildingId && cachedBuildingDetail?.id === buildingId) {
+      if (
+        [SidebarContextType.VIEW_BUILDING, SidebarContextType.UPDATE_BUILDING].includes(
+          sideBarContext,
+        )
+      ) {
+        const newContext = disabled
+          ? SidebarContextType.VIEW_BUILDING
+          : SidebarContextType.UPDATE_BUILDING;
+        newContext !== sideBarContext && setSideBarContext(newContext);
+      }
+    } else if (!!cachedBuildingDetail && !cachedBuildingDetail?.id) {
+      setSideBarContext(SidebarContextType.ADD_BUILDING);
```

Comment:
technically we are adding a building even though we are passing initial building data.